library/src/main/java/bio/terra/stairway/Control.java

```diff
@@ -0,0 +1,583 @@
+package bio.terra.stairway;
+
+import bio.terra.stairway.exception.DatabaseOperationException;
+import bio.terra.stairway.exception.FlightException;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import javax.sql.DataSource;
+
+@SuppressFBWarnings(
+    value = "RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE",
+    justification = "Spotbugs doesn't understand resource try construct")
+public class Control {
+  private final Stairway stairway;
+  private final DataSource dataSource;
+  private final FlightDao flightDao;
+  private final StairwayInstanceDao stairwayInstanceDao;
+
+  Control(
+      Stairway stairway,
+      DataSource dataSource,
+      FlightDao flightDao,
+      StairwayInstanceDao stairwayInstanceDao) {
+    this.stairway = stairway;
+    this.dataSource = dataSource;
+    this.flightDao = flightDao;
+    this.stairwayInstanceDao = stairwayInstanceDao;
+  }
+
+  // -- Flight methods --
+
+  private static final String flightSelectFrom =
+      " flightid, submit_time, class_name, completed_time, status,"
+          + "serialized_exception, stairway_id FROM flight ";
+
+  private static final String flightOrderPage =
+      " ORDER BY submit_time DESC OFFSET :offset LIMIT :limit";
+
+  private int countQuery(NamedParameterPreparedStatement statement) throws SQLException {
+    int flightCount = 0;
+    try (ResultSet rs = statement.getPreparedStatement().executeQuery()) {
+      while (rs.next()) {
+        flightCount = rs.getInt("total");
+      }
+    }
+    return flightCount;
+  }
+
+  public int countFlights(String status) throws SQLException {
```

Comment:
nit: any particular reason to take a String instead of a FlightStatus and leave parsing FlightStatus to clients? Ditto others. if it's particularly helpful, we could provide a static conversion or throw utility and deduplicate some code.

Reply:
Good idea